### PR TITLE
pkg/openthread: bump version to 20200818 + adapt to use CMake to configure the build

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR bumps the version of openthread to the [latest 20200818 release](https://github.com/openthread/openthread/releases/tag/thread-reference-20200818). Since this release can now be built using CMake, this PR also adapt the package to cmake instead of autotools to build it: this enables out-of-source build like all other packages (except micropython).

### Testing procedure

- A green CI
- Follow the readme of `examples/openthread` and ensure that it's still functional

### Issues/PRs references

None

### CODE REVIEW

**Comment by jia200x** [comment]
I don't have too much time to test, but it makes sense. I will see if I can give it a try during the week

File: pkg/openthread/contrib/netdev/openthread_netdev.c [hunk 1]
```diff
@@ -94,10 +94,7 @@ static void *_openthread_event_loop(void *arg)
     /* init OpenThread */
     sInstance = otInstanceInitSingle();
 
-    /* enable OpenThread UART */
-    otPlatUartEnable();
```
Comment (aabadie) [inline pkg/openthread/contrib/netdev/openthread_netdev.c:98]: Maybe this should stay if not using the CLI (because the CLI enables UART internally)

**Comment by fjmolinas** [comment]
@aabadie can you provide testing output for this otherwise?

**Comment by aabadie** [comment]
<details><summary>This is the output I have on one iot-lab m3:<summary>

```
ipaddr
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:c000
fdde:ad00:beef:0:8835:64e6:2e3c:4625
fe80:0:0:0:a021:6a42:9ce3:7a6
Done
> ping fdde:ad00:beef:0:70f:d972:9112:3e6f
ping fdde:ad00:beef:0:70f:d972:9112:3e6f
Done
> coap
coap
help
delete
get
parameters
post
put
resource
set
start
stop
Error 7: InvalidArgs
> coap help
coap help
help
delete
get
parameters
post
put
resource
set
start
stop
Done
> coap help
coap help
help
delete
get
parameters
post
put
resource
set
start
stop
Done
> help
help
bufferinfo
channel
child
childip
childmax
childtimeout
coap
contextreusedelay
counters
dataset
delaytimermin
discover
dns
eidcache
eui64
log
extaddr
extpanid
factoryreset
help
ifconfig
ipaddr
ipmaddr
joinerport
keysequence
leaderdata
leaderpartitionid
leaderweight
mac
masterkey
mode
neighbor
netdata
netdatashow
netstat
networkdiagnostic
networkidtimeout
networkname
panid
parent
parentpriority
ping
pollperiod
promiscuous
preferrouterid
pskc
rcp
releaserouterid
reset
rloc16
router
routerdowngradethreshold
routereligible
routerselectionjitter
routerupgradethreshold
scan
singleton
state
thread
txpower
udp
unsecureport
version
Done
> ping fdde:ad00:beef:0:70f:d972:9112:3e6f
ping fdde:ad00:beef:0:70f:d972:9112:3e6f
Done
> 16 bytes from fdde:ad00:beef:0:70f:d972:9112:3e6f: icmp_seq=2 hlim=64 time=15ms

ping fdde:ad00:beef:0:70f:d972:9112:3e6f
ping fdde:ad00:beef:0:70f:d972:9112:3e6f
Done
> 16 bytes from fdde:ad00:beef:0:70f:d972:9112:3e6f: icmp_seq=3 hlim=64 time=12ms


help
help
bufferinfo
channel
child
childip
childmax
childtimeout
coap
contextreusedelay
counters
dataset
delaytimermin
discover
dns
eidcache
eui64
log
extaddr
extpanid
factoryreset
help
ifconfig
ipaddr
ipmaddr
joinerport
keysequence
leaderdata
leaderpartitionid
leaderweight
mac
masterkey
mode
neighbor
netdata
netdatashow
netstat
networkdiagnostic
networkidtimeout
networkname
panid
parent
parentpriority
ping
pollperiod
promiscuous
preferrouterid
pskc
rcp
releaserouterid
reset
rloc16
router
routerdowngradethreshold
routereligible
routerselectionjitter
routerupgradethreshold
scan
singleton
state
thread
txpower
udp
unsecureport
version
Done
> 
coap start
coap start
Done
> coap resource test-resource
coap resource test-resource
Done
> ipaddr
ipaddr
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:c000
fdde:ad00:beef:0:8835:64e6:2e3c:4625
fe80:0:0:0:a021:6a42:9ce3:7a6
Done
> coap request from fdde:ad00:beef:0:70f:d972:9112:3e6f GET
coap response sent
coap request from fdde:ad00:beef:0:70f:d972:9112:3e6f PUT with payload: 7061796c6f6164
coap response sent
coap request from fdde:ad00:beef:0:70f:d972:9112:3e6f PUT with payload: 746f746f
coap response sent
coap request from fdde:ad00:beef:0:70f:d972:9112:3e6f PUT with payload: 3432
coap response sent
^C
abadie@saclay:~$ logout
```

</details>

I don't have anymore the output of the other M3, but it was correct AFAICT.

**Comment by fjmolinas** [comment]
@aabadie any change you could test as requested?

**Comment by aabadie** [comment]
Some output with an experiment on Grenoble with several M3 nodes:

<details>

```
$ abadie@grenoble:~$ serial_aggregator 
1637589803.167308;Aggregator started
1637589811.902112;m3-22;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.902349;m3-22;This is an OpenThread example
1637589811.903003;m3-22;Current panid: 0xbeef
1637589811.904581;m3-12;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.905654;m3-12;This is an OpenThread example
1637589811.905782;m3-12;Current panid: 0xbeef
1637589811.909235;m3-2;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.910171;m3-2;This is an OpenThread example
1637589811.910401;m3-2;Current panid: 0xbeef
1637589811.934009;m3-17;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.934252;m3-17;This is an OpenThread example
1637589811.934960;m3-17;Current panid: 0xbeef
1637589811.936574;m3-42;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.937298;m3-42;This is an OpenThread example
1637589811.937464;m3-42;Current panid: 0xbeef
1637589811.940520;m3-34;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.940720;m3-34;This is an OpenThread example
1637589811.941351;m3-34;Current panid: 0xbeef
1637589811.948438;m3-30;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.948977;m3-30;This is an OpenThread example
1637589811.949983;m3-30;Current panid: 0xbeef
1637589811.994180;m3-7;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589811.995343;m3-7;This is an OpenThread example
1637589811.995571;m3-7;Current panid: 0xbeef
1637589812.016150;m3-38;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589812.016355;m3-38;This is an OpenThread example
1637589812.016969;m3-38;Current panid: 0xbeef
1637589812.035218;m3-26;main(): This is RIOT! (Version: 2022.01-devel-649-gd77975-pr/pkg/openthread_20200818)
1637589812.036099;m3-26;This is an OpenThread example
1637589812.036236;m3-26;Current panid: 0xbeef
# Print the state of all nodes, there's only one leader, some routers and child
state
1637589867.583758;m3-12;state
1637589867.583983;m3-12;router
1637589867.584143;m3-34;state
1637589867.584251;m3-34;router
1637589867.584345;m3-34;Done
1637589867.584718;m3-17;state
1637589867.584846;m3-17;leader
1637589867.585105;m3-17;Done
1637589867.585236;m3-2;state
1637589867.585394;m3-2;router
1637589867.585468;m3-2;Done
1637589867.585561;m3-22;state
1637589867.585649;m3-30;state
1637589867.585722;m3-30;child
1637589867.585866;m3-30;Done
1637589867.586070;m3-38;state
1637589867.586162;m3-38;child
1637589867.586241;m3-38;Done
1637589867.586336;m3-42;state
1637589867.586413;m3-42;child
1637589867.586487;m3-42;Done
1637589867.586643;m3-12;Done
1637589867.586759;m3-22;child
1637589867.586846;m3-22;Done
1637589867.586944;m3-26;state
1637589867.587022;m3-26;router
1637589867.587098;m3-26;Done
1637589867.587193;m3-7;state
1637589867.587268;m3-7;router
1637589867.587341;m3-7;Done
# List the IP addresses of each nodes
ipaddr
1637589928.143726;m3-12;> ipaddr
1637589928.144508;m3-12;fdde:ad00:beef:0:0:ff:fe00:400
1637589928.144915;m3-17;> ipaddr
1637589928.145136;m3-17;fdde:ad00:beef:0:0:ff:fe00:fc00
1637589928.145312;m3-17;fdde:ad00:beef:0:0:ff:fe00:5800
1637589928.145668;m3-2;> ipaddr
1637589928.145945;m3-22;> ipaddr
1637589928.146116;m3-26;> ipaddr
1637589928.146231;m3-26;fdde:ad00:beef:0:0:ff:fe00:6000
1637589928.146356;m3-26;fdde:ad00:beef:0:7a6f:9e76:8c8c:fb25
1637589928.146429;m3-26;fe80:0:0:0:9036:a5e1:a0a1:651a
1637589928.146550;m3-26;Done
1637589928.146652;m3-34;> ipaddr
1637589928.146820;m3-34;fdde:ad00:beef:0:0:ff:fe00:c00
1637589928.146929;m3-34;fdde:ad00:beef:0:e007:59b4:a1cd:bafc
1637589928.146998;m3-34;fe80:0:0:0:3499:31ed:65be:aba1
1637589928.147066;m3-34;Done
1637589928.147208;m3-38;> ipaddr
1637589928.147285;m3-38;fdde:ad00:beef:0:0:ff:fe00:4000
1637589928.147390;m3-38;fdde:ad00:beef:0:f16c:5ed3:e1e2:d230
1637589928.147497;m3-38;fe80:0:0:0:80e1:7bf1:ad7b:132f
1637589928.147610;m3-38;Done
1637589928.147786;m3-7;> ipaddr
1637589928.147951;m3-7;fdde:ad00:beef:0:0:ff:fe00:cc00
1637589928.148032;m3-7;fdde:ad00:beef:0:fdb0:4fb9:9896:8e3b
1637589928.148149;m3-7;fe80:0:0:0:54a9:cd8c:4322:85e3
1637589928.148230;m3-7;Done
1637589928.148386;m3-12;fdde:ad00:beef:0:746d:761a:c012:ec7e
1637589928.148461;m3-12;fe80:0:0:0:d8c6:f947:48e8:3a3f
1637589928.148529;m3-12;Done
1637589928.148625;m3-17;fdde:ad00:beef:0:3872:ef0c:e27:2db
1637589928.148692;m3-17;fe80:0:0:0:6479:52ac:ff2:a8b2
1637589928.148739;m3-17;Done
1637589928.148792;m3-2;fdde:ad00:beef:0:0:ff:fe00:e800
1637589928.148832;m3-2;fdde:ad00:beef:0:58f7:e012:4943:8d
1637589928.148871;m3-2;fe80:0:0:0:34cf:5229:d4a7:453b
1637589928.148909;m3-2;Done
1637589928.148977;m3-22;fdde:ad00:beef:0:0:ff:fe00:800
1637589928.149038;m3-22;fdde:ad00:beef:0:de49:abd1:2331:edb4
1637589928.149096;m3-22;fe80:0:0:0:a409:d573:de9c:fc62
1637589928.149192;m3-22;Done
1637589928.149267;m3-30;> ipaddr
1637589928.149344;m3-30;fdde:ad00:beef:0:0:ff:fe00:c03
1637589928.149408;m3-30;fdde:ad00:beef:0:55f:8f06:8092:35c8
1637589928.149468;m3-30;fe80:0:0:0:ec09:a34c:e703:11d9
1637589928.149525;m3-30;Done
1637589928.149614;m3-42;> ipaddr
1637589928.149740;m3-42;fdde:ad00:beef:0:0:ff:fe00:1800
1637589928.149814;m3-42;fdde:ad00:beef:0:c55d:67bf:b45a:43e8
1637589928.149877;m3-42;fe80:0:0:0:908b:5525:7a95:d0ff
1637589928.149937;m3-42;Done
# Ping node m3-42 from m3-2
m3-2;ping fdde:ad00:beef:0:0:ff:fe00:1800
1637589977.097016;m3-2;> ping fdde:ad00:beef:0:0:ff:fe00:1800
1637589977.097863;m3-2;Done
1637589977.121416;m3-2;> 16 bytes from fdde:ad00:beef:0:0:ff:fe00:1800: icmp_seq=1 hlim=64 time=24ms
# Ping node m3-7 from m3-42
m3-42;ping fdde:ad00:beef:0:0:ff:fe00:cc00
1637589994.194736;m3-42;> ping fdde:ad00:beef:0:0:ff:fe00:cc00
1637589994.194967;m3-42;Done
1637589994.222758;m3-42;> 16 bytes from fdde:ad00:beef:0:0:ff:fe00:cc00: icmp_seq=1 hlim=64 time=28ms
```

</details>

**Comment by aabadie** [comment]
I pushed a small change to support CMake 3.20

**Review by benpicco — APPROVED** [review]

**Comment by aabadie** [comment]
Thanks !

**Pull request merged**